### ISSUE TITLE
Help generating mocks for generics

### ISSUE THREAD

**Issue by btomasini** [golang/mock#667]
I am trying to execute `mockgen` in source mode against some code which contains generics.

I the following error:

```
2022/07/27 17:19:21 Loading input failed: don't know how to mock method of type *ast.IndexListExpr
repository/repository.go:3: running "mockgen": exit status 1
```

My guess is that I need to use a fork which contains support for generics.  As such, I have tried to replace gomock with the following branch in my `go.mod`:

```
replace (
        github.com/golang/mock => github.com/bradleygore/gomock v1.6.1-0.20220713191646-ac74355d2d11
        github.com/golang/mock/mockgen => github.com/bradleygore/gomock/mock/mockgen v1.6.1-0.20220713191646-ac74355d2d11
)
```

But `mockgen` still errors out.  I think I may not be installing the right version of mockgen.  My generate executes the following:

```gp
//go:generate go install github.com/golang/mock/mockgen@v1.6.0
//go:generate mockgen -source=client.go -destination=./mock_repository/mock_client.go
```

Any help would be appreciated.  I am not sure if my problem is one of managing a forked go module properly or with mockgen itself.

Thank you.

**Comment by btomasini** [comment]
I hope this is somehow resolved by #663

**Comment by codyoss** [comment]
Some general generic use-cases have been resolved on HEAD but not released. Follow the milestone for progress towards the release.
